addons/repair/models/sale_order.py

```diff
@@ -85,6 +85,19 @@ def _action_launch_stock_rule(self, previous_product_uom_qty=False):
     def _create_repair_order(self):
         new_repair_vals = []
         for line in self:
+            # In instances of one line with products individually tracked via serial, create multiple ROs
+            if line.product_id.product_tmpl_id.tracking == 'serial':
+                for _ in range(int(line.product_uom_qty)):
+                    new_repair_vals.append({
+                        'state': 'confirmed',
+                        'partner_id': line.order_id.partner_id.id,
+                        'sale_order_id': line.order_id.id,
+                        'sale_order_line_id': line.id,
+                        'picking_type_id': line.order_id.warehouse_id.repair_type_id.id,
+                        'product_id': line.product_id.id,
+                        'product_uom': line.product_uom.id,
+                    })
+                continue
```

Comment:
Few comments:
\- When dealing with SN, we consider the qties as integer
\- We should first check the two conditions below, in case we need to skip the RO creation
\- I think `continue` is not always that readable
\- We could factorize the RO values a bit

So, something like
```diff
diff --git a/addons/repair/models/sale_order.py b/addons/repair/models/sale_order.py
index d565d93a7c79..3d22a4439fc9 100644
--- a/addons/repair/models/sale_order.py
+++ b/addons/repair/models/sale_order.py
@@ -85,20 +85,6 @@ class SaleOrderLine(models.Model):
     def _create_repair_order(self):
         new_repair_vals = []
         for line in self:
-            # In instances of one line with products individually tracked via serial, create multiple ROs
-            if line.product_id.product_tmpl_id.tracking == 'serial':
-                for _ in range(int(line.product_uom_qty)):
-                    new_repair_vals.append({
-                        'state': 'confirmed',
-                        'partner_id': line.order_id.partner_id.id,
-                        'sale_order_id': line.order_id.id,
-                        'sale_order_line_id': line.id,
-                        'picking_type_id': line.order_id.warehouse_id.repair_type_id.id,
-                        'product_id': line.product_id.id,
-                        'product_uom': line.product_uom.id,
-                    })
-                continue
-            # One RO for each line with at least a quantity of 1, quantities > 1 don't create multiple ROs
             if any(line.id == ro.sale_order_line_id.id for ro in line.order_id.sudo().repair_order_ids) and float_compare(line.product_uom_qty, 0, precision_rounding=line.product_uom.rounding) > 0:
                 binded_ro_ids = line.order_id.sudo().repair_order_ids.filtered(lambda ro: ro.sale_order_line_id.id == line.id and ro.state == 'cancel')
                 binded_ro_ids.action_repair_cancel_draft()
@@ -106,20 +92,32 @@ class SaleOrderLine(models.Model):
                 continue
             if not line.product_template_id.sudo().create_repair or line.move_ids.sudo().repair_id or float_compare(line.product_uom_qty, 0, precision_rounding=line.product_uom.rounding) <= 0:
                 continue
+
             order = line.order_id
-            new_repair_vals.append({
+            default_repair_vals = {
                 'state': 'confirmed',
                 'partner_id': order.partner_id.id,
                 'sale_order_id': order.id,
                 'sale_order_line_id': line.id,
                 'picking_type_id': order.warehouse_id.repair_type_id.id,
-            })
-            if line.product_template_id.type in ('consu', 'product'):
-                new_repair_vals[-1].update({
+            }
+            if line.product_id.tracking == 'serial':
+                vals = {
+                    **default_repair_vals,
+                    'product_id': line.product_id.id,
+                    'product_qty': 1,
+                    'product_uom': line.product_uom.id,
+                }
+                new_repair_vals.extend([vals] * int(line.product_uom_qty))
+            elif line.product_id.type in ('consu', 'product'):
+                new_repair_vals.append({
+                    **default_repair_vals,
                     'product_id': line.product_id.id,
                     'product_qty': line.product_uom_qty,
                     'product_uom': line.product_uom.id,
                 })
+            else:
+                new_repair_vals.append(default_repair_vals.copy())
 
         if new_repair_vals:
             self.env['repair.order'].sudo().create(new_repair_vals)
```